packages/shell-evaluator/src/shell-evaluator.ts

```diff
@@ -93,7 +93,8 @@ class ShellEvaluator<EvaluationResultType = ShellResult> {
     }
 
     if (this.exposeAsyncRewriter) {
-      (context as any).__asyncRewrite = () => this.asyncWriter.process(input);
+      (context as any).__asyncRewrite = (input: string) =>
+        this.asyncWriter.process(input);
```

Comment:
[nit, fine as is], how do you feel about renaming `input` to make sure we don't confuse it with the innerEval input?
```suggestion
      (context as any).__asyncRewrite = (rewriteInput: string) =>
        this.asyncWriter.process(rewriteInput);
```

Reply:
sgtm!